src/nvim/api/extmark.c

```diff
@@ -117,6 +119,10 @@ static Array extmark_to_array(const ExtmarkInfo *extmark, bool id, bool add_dict
   if (add_dict) {
     Dictionary dict = ARRAY_DICT_INIT;
 
+    String ns = copy_string(map_get(handle_T, String)(&namespaces, (handle_T)extmark->ns_id), NULL);
+    PUT(dict, "ns", STRING_OBJ(ns));
```

Comment:
if we are anyway adding the functionality to lookup namespaces per handle, adding this string here is redundant (and avoiding it would save quite some allocations)

Reply:
Yeah I was thinking of filtering by name but yeah mapping id's to name in lua is fine.